### PR TITLE
feat: retrieve users' notifications

### PR DESCRIPTION
**Pull request by zee-ham-su**
## What does the PR do?
This PR addresses issue of implementing the functionality to retrieve notifications for authenticated users from the database.  The endpoint returns errors if:

- [x] User is not logged in.
- [x] User does not exist.
- [x] Validation errors on the user input is done

## How should this be manually tested?
1. Login to a user account, and get the access token
2. You can send a GET request to the following endpoint:
GET /api/v1/notifications

### Success Response
on a successful execution, you will receive the following response structure:
```
{
    "status": "success",
    "status_code": 200,
    "message": "Notifications retrieved successfully",
    "data": {
        "total_notification_count": 2,
        "total_unread_notification_count": 2,
        "notifications": [
            {
                "notification_id": "0bad0dae-a11a-41e4-beb7-230808c1f3d5",
                "is_read": false,
                "message": "Notification 1 for John",
                "created_at": "2024-07-28T19:29:13.049Z"
            },
            {
                "notification_id": "f16e9a12-b428-4bcc-b9d9-e560758e15c8",
                "is_read": false,
                "message": "Notification 2 for John",
                "created_at": "2024-07-28T19:29:13.049Z"
            }
        ]
    }
}
```
### "Invalid Input" Response
If the request is invalid, you will receive the following response:
```
{
  "status_code": 400,
  "message": "Invalid request.",
  "data": null
}

```
### Unauthorized Response
```
{
  "status_code": 401,
  "status": "unauthorized",
  "message": "Unauthorized. Please log in."
}
```

### "Internal Server Error" Response

```
{
  "status_code": 500,
  "status": "Internal server error",
  "message": "Failed to retrieve notifications.",
  "data": null
}

```

## Task
- [x]  Implemented GET /api/v1/notifications endpoint in notifications.controller.ts
- [x] Added method to fetch notifications in the NotificationsService.
- [x] Tested endpoint functionality to ensure it handles various scenarios correctly
- [x] Added appropriate HTTP status codes and response messages for edge cases. 
- [x] Added unit tests for the NotificationsService.
- [x] Updated API documentation to reflect the new endpoint

Reference Issue: [Link to Issue](https://linear.app/team-bingo/issue/BAC2-42/retrieve-all-notifications-for-a-user)

### CODE REVIEW

File: src/modules/invite/tests/invite.service.spec.ts [hunk 1]
```diff
@@ -48,6 +49,7 @@ describe('InviteService', () => {
       secret: 'secret',
       is_2fa_enabled: true,
       testimonials: [],
+      notifications: [] as Notification[],
```
Comment (CFCIfe) [inline src/modules/invite/tests/invite.service.spec.ts:52]:
```suggestion
notifications: [] as Notification[]
```

File: src/modules/notifications/notifications.controller.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+@ApiBearerAuth()
+@ApiTags('Notifications')
+@Controller('notifications')
+export class NotificationsController {
+  constructor(private readonly notificationsService: NotificationsService) {}
+
+  @Get()
+  async getNotifications(@Req() req: { user: UserPayload }) {
+    const userId = req.user.id;
+    try {
```
Comment (CFCIfe) [inline src/modules/notifications/notifications.controller.ts:15]: move the `try...catch` to your service

File: src/modules/notifications/notifications.controller.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { UserPayload } from '../user/interfaces/user-payload.interface';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+
+@ApiBearerAuth()
+@ApiTags('Notifications')
+@Controller('notifications')
+export class NotificationsController {
+  constructor(private readonly notificationsService: NotificationsService) {}
+
+  @Get()
```
Comment (CFCIfe) [inline src/modules/notifications/notifications.controller.ts:12]: Add swagger documentation for success and error response

File: src/modules/notifications/notifications.controller.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            created_at: notification.created_at,
+          })),
+        },
+      };
+    } catch (error) {
+      return {
+        status: 'error',
+        message: 'Failed to retrieve notifications.',
+        status_code: 500,
+        data: null,
```
Comment (CFCIfe) [inline src/modules/notifications/notifications.controller.ts:38]: Take this off. We don't use this value.

Also use `throw new InternalServerErrorException` to wrap your return statement

File: src/modules/organisations/tests/mocks/organisation.mock.ts [hunk 5]
```diff
@@ -59,6 +60,7 @@ export const createMockOrganisation = (): Organisation => {
     user_type: UserType.ADMIN,
     secret: 'secret',
     is_2fa_enabled: false,
+    notifications: [] as Notification[],
```
Comment (CFCIfe) [inline src/modules/organisations/tests/mocks/organisation.mock.ts:63]:
```suggestion
notifications: []
```

**Review by CFCIfe — COMMENTED** [review]
Review done
